File: tools/wax/MxStat/tags/Release_1.0/MxStat/Entities/EnWaxDate.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MxStat.Entities
{
    public class EnWaxDate
    {
        long mWaxDate;

        public EnWaxDate(DateTime date)
        {
            setDate(date);
        }

        public EnWaxDate()
        {
        }

        public long Value
        {
            get
            {
                return mWaxDate;
            }
            set
            {
                mWaxDate = value;
            }

        }

        public void setDate(DateTime date)
        {
            long diffDays;


            //BasisDatum COleDateTime
            DateTime oleDate = new DateTime(1899,12,30);

            //Tage zum Basisdatum 
            TimeSpan span = date - oleDate;
            diffDays = span.Days;

            //umrechnen in Wax-Datum
            mWaxDate = diffDays * 10000;
        }


        public DateTime getDate()
        {

            //auf diesem Datum basiert das COleDateTime Objekt. 
            DateTime date = new DateTime(1899,12,30);

            //Wax Datum ins COleDateTime Basisdatum umrechnen
            long waxDate = mWaxDate / 10000;

            TimeSpan span = new TimeSpan((int)waxDate, 0, 0, 0);

            //Tage anhängen
            date += span;

            return date;
        }

        /// <summary>
        /// Auf Tagesanfang abrunden
        /// </summary>
        public void abrunden()
        {
            mWaxDate = (mWaxDate / 10000) * 10000;
        }

        /// <summary>
        /// auf Tagesende aufrunden
        /// </summary>
        public void aufrunden()
        {
            mWaxDate = (mWaxDate / 10000) * 10000 + 9999;
        }

        public override string ToString()
        {
            return mWaxDate.ToString();
        }
    }
}

File: tools/wax/waxstat/EnWaxDate.cpp
#include ".\enwaxdate.h"

EnWaxDate::EnWaxDate(void)
{
	m_waxDate = 0;
}

EnWaxDate::~EnWaxDate(void)
{
}

///Datum in Wax-Format setzen
void EnWaxDate::setWaxDate(long date)
{
	m_waxDate = date;
}

long EnWaxDate::getWaxDate()
{
	return m_waxDate;
}

/**
Datum in QDate Format setzen.
Wird intern in das Wax-Datum umgerechnet
*/
void EnWaxDate::setDate(const QDate &date)
{
	long baseQDate,
		 diffDays;


	QDate oleDate;

	//BasisDatum COleDateTime
	oleDate.setYMD(1899,12,30);

	//Tage zum Basisdatum von COledateTime
	diffDays = oleDate.daysTo(date);

	//umrechnen in Wax-Datum
	m_waxDate = diffDays * 10000;

}

/**
Das Wax-Datumsformat in QDate zurückgeben
*/
QDate EnWaxDate::getDate()
{
	QDate date;

	//Wax Datum ins COleDateTime Basisdatum umrechnen
	long waxDate = m_waxDate / 10000;

	//auf diesem Datum basiert das COleDateTime Objekt. 
	date.setYMD(1899,12,30);

	//Tage anhängen
	date = date.addDays(waxDate);


	return date;
}